_rules/heading-descriptive-b49b2e.md

```diff
@@ -30,7 +30,11 @@ This rule applies to any element with the [semantic role][] of heading that is e
 
 ## Expectation
 
-Each target element describes the topic or purpose of its [section of the content][].
+Each target element describes the topic or purpose of the first [palpable content][] which is:
+
+- after the target, in tree order in the [flat tree][]; and
+- [visible][], if the target is [visible][]; and
+- [included in the accessibility tree][], if the target is [included in the accessibility tree][].
```

Comment:
This strongly suggest that this rule should be split in two, one for each case…

Reply:
Not if those 'and' conditional are intentional. I read this as:

If the target is visible and included in the accessibility tree, then it should describe the first palpable content that is visible and included in the accessibility tree that comes after the target in the flat tree.

However, what about non-visible headings describing visible content, where both are included in the accessibility tree? This is not an unusual use case, eg. a news front page with a hidden heading 'Top headlines'. The heading isn't needed visually as that is usually apparent from other cues.

Reply:
If it is both visible and included in the accessibility tree, then it is "visible or included in the accessibility tree" (the or is inclusive by default). And then it matches both the expectations and need to describe both visible and accessible content (which may be the same or different palpable content).